src/lxc/lxccontainer.c

```diff
@@ -4659,6 +4680,131 @@ static bool do_lxcapi_restore(struct lxc_container *c, char *directory, bool ver
 
 WRAP_API_2(bool, lxcapi_restore, char *, bool)
 
+static int do_lxcapi_mount(struct lxc_container *c,
+			const char *source, const char *target,
+			const char *filesystemtype, unsigned long mountflags,
+			const void *data, struct lxc_mount *mnt) {
+	int ret;
+	int len;
+	int fd;
+	int status;
+	char* pref = NULL;
+	char* temp;
+	char *suff;
+	char *path;
+	pid_t pid;
+	pid_t init_pid;
+	struct stat sb;
+
+	// Create a name template
+	// Do it here to be able to use temp in both parent and child
+	pref = getenv("LXC_DIR");
+	if (pref == NULL)
+		pref = "/var/lib/lxc";
+
+	len = strlen(pref) + strlen("shmounts") + strlen(c->name) + strlen("lxcmountXXXXXX") + 4;
+	temp = (char *)malloc(len);
+	ret = snprintf(temp, len, "%s/shmounts/%s/lxcmountXXXXXX", pref, c->name);
+	if (ret < 0 || ret >= len) {
+		ERROR("Error writing shmounts tempdir name");
+		return -1;
+	}
```

Comment:
The path is wrong. You do not want to create a temporary mountpoint on the host at `/var/lib/lxc/shmounts/<container-name>/<tmp>` but at `conf->shmounts_path/<tmp>`. :)